File: 백준/Bronze/16770. The Bucket List/The Bucket List.py
import sys
input = lambda: sys.stdin.readline().rstrip()


def main():
    N = int(input())
    buckets = [0]*1001
    for n in range(1, N+1):
        s, t, b = map(int, input().split())
        for i in range(s, t+1):
            buckets[i] += b
    print(max(buckets))


if __name__ == '__main__':
    main()

File: 백준/Bronze/16770. The Bucket List/The Bucket List.c
#include <stdio.h>

int main() {
    int N;
    scanf("%d", &N);
    int buckets[1001] = {0};
    for (int n = 0; n < N; n++) {
        int s, t, b;
        scanf("%d %d %d", &s, &t, &b);
        for (int i = s; i <= t; i++) {
            buckets[i] += b;
        }
    }
    int max = 0;
    for (int i = 0; i < 1001; i++) {
        if (buckets[i] > max) {
            max = buckets[i];
        }
    }
    printf("%d\n", max);
    return 0;
}